python/lsst/analysis/tools/interfaces.py

```diff
@@ -420,6 +420,9 @@ def getOutputNames(self) -> Iterable[str]:
         """
         outNames = tuple(self.produce.getOutputNames())
         if outNames:
-            return outNames
+            return (f"{self.identity or ''}_{name}" for name in outNames)
         else:
-            return (f"{{band}}_{self.identity or ''}",)
+            if self.parameterizedBand:
+                return (f"{{band}}_{self.identity or ''}",)
```

Comment:
As above, this could end up returning `{band}_`, but, happy to ignore this for the time being.